apps/base-docs/docs/building-with-base/guides/onchain-generative-nfts.md

```diff
@@ -0,0 +1,876 @@
+---
+title: 'Onchain Generative Art NFTs'
+description: Learn how to make complex nfts that are procedurally generated and have onchain metadata and images.
+keywords:
+  [
+    Solidity,
+    ERC-721,
+    token,
+    NFT,
+    SVG,
+    generative art,
+    onchain metadata,
+    nft metadata,
+    onchain images,
+    interfaces,
+    pseudorandom numbers,
+    deterministic randomness,
+  ]
+hide_table_of_contents: false
+---
+
+Many NFTs are dependent on offchain metadata and images. Some use immutable storage locations, such as [IPFS]. Others use traditional web locations, and many of these allow the owner of the contract to modify the URL returned by a contract when a site or user attempts to retrieve the location of the token art and metadata. This power isn't inherently bad, because we probably want someone to be able to fix the contract if the storage location goes down. However, it does introduce a requirement to trust the contract owner.
```

Comment:
Is IPFS supposed to link somewhere? Do we need the [ ] if not?

Reply:
Do we have an IPFS page?

Reply:
It just goes to IPFS's home page.  We use reference-style links so that the raw markdown is more readable when writing and editing